raiden/storage/migrations/v19_to_v20.py

```diff
@@ -0,0 +1,168 @@
+
+def _add_onchain_locksroot_to_channel_settled_state_changes(
+        raiden: RaidenService,
+        storage: SQLiteStorage,
+) -> None:
+    """ Adds `our_onchain_locksroot` and `partner_onchain_locksroot` to
+    ContractReceiveChannelSettled. """
+    batch_size = 50
+    batch_query = storage.batch_query_state_changes(
+        batch_size=batch_size,
+        filters=[
+            ('_type', 'raiden.transfer.state_change.ContractReceiveChannelSettled'),
+        ],
+    )
+    for state_changes_batch in batch_query:
+        updated_state_changes = []
+        for state_change in state_changes_batch:
+            data = json.loads(state_change.data)
+            msg = 'v18 state changes cant contain our_onchain_locksroot'
+            assert 'our_onchain_locksroot' not in data, msg
+
+            msg = 'v18 state changes cant contain partner_onchain_locksroot'
+            assert 'partner_onchain_locksroot' not in data, msg
+
+            state_change_data = json.loads(state_change.data)
+            token_network_identifier = state_change_data['token_network_identifier']
+            channel_identifier = state_change_data['channel_identifier']
+
+            channel_new_state_change = _find_channel_new_state_change(
+                storage=storage,
+                token_network_address=token_network_identifier,
+                channel_identifier=channel_identifier,
+            )
+
+            if not channel_new_state_change.data:
+                raise RaidenUnrecoverableError(
+                    'Could not find the state change for channel {channel_identifier}, '
+                    'token network address: {token_network_identifier} being created. ',
+                )
+
+            channel_state_data = json.loads(channel_new_state_change.data)
+            new_channel_state = channel_state_data['channel_state']
+            our_locksroot, partner_locksroot = get_onchain_locksroots(
+                raiden=raiden,
+                token_network_address=token_network_identifier,
+                channel_identifier=channel_identifier,
+                participant1=new_channel_state['our_state']['address'],
+                participant2=new_channel_state['partner_state']['address'],
+                block_identifier='latest',
+            )
+
+            data['our_onchain_locksroot'] = our_locksroot
+            data['partner_onchain_locksroot'] = partner_locksroot
+
+            updated_state_changes.append((
+                json.dumps(data),
+                state_change.state_change_identifier,
+            ))
+        storage.update_state_changes(updated_state_changes)
+
+
+def _add_onchain_locksroot_to_snapshot(
+        raiden: RaidenService,
+        storage: SQLiteStorage,
+        raw_snapshot: Dict[str, Any],
+):
+    """
+    Add `onchain_locksroot` to each NettingChannelEndState
+    """
+    snapshot = json.loads(raw_snapshot)
+
+    for payment_network in snapshot.get('identifiers_to_paymentnetworks', {}).values():
```

Comment:
nitpick: consistency. Please use either `{}` or `dict()` to specify an empty dict in this function. Same goes for list.